airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -115,24 +116,40 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         logger.info(f"Using start_date: {config['start_date']}, end_date: {config['end_date']}")
 
         auth = self.get_authenticator(config)
-        streams = []
-        for stream in [
-            Annotations(authenticator=auth, **config),
-            Cohorts(authenticator=auth, **config),
-            Funnels(authenticator=auth, **config),
-            Revenue(authenticator=auth, **config),
-            CohortMembers(authenticator=auth, **config),
-            Engage(authenticator=auth, **config),
+        streams = [
             Export(authenticator=auth, **config),
-        ]:
+        ]
+
+        # we only make calls to one stream of each API type to save time, based on the assumption that
+        # if one stream of the API type is available, all others are available as well and vice versa
+
+        streams_by_api_types = [
+            [
+                Annotations,
+            ],
+            [Engage, Cohorts, CohortMembers, Funnels, Revenue],
+        ]
+        stream_kwargs = {"authenticator": auth, **config}
+        if not os.environ.get("SLEEP_BETWEEN_CHECK_AND_DISCOVER_REQUESTS", False):
+            # set reqs_per_hour_limit = 0 to save time for discovery
+            # We preserve sleeping between requests in case this is a running acceptance test.
```

Comment:
Every API call is followed by sleeping ~60 seconds. We can wait when running tests, but we shouldn't do it whe the connector is in use, as it can lead to timeouts. -- Extended the comment in the code explaining this